crates/katana/core/src/service/messaging/ethereum.rs

```diff
@@ -76,10 +76,10 @@ impl EthereumMessaging {
         &self,
         from_block: u64,
         to_block: u64,
-    ) -> MessengerResult<HashMap<u64, Vec<Log>>> {
+    ) -> MessengerResult<Vec<Log>> {
         trace!(target: LOG_TARGET, from_block = ?from_block, to_block = ?to_block, "Fetching logs.");
 
-        let mut block_to_logs: HashMap<u64, Vec<Log>> = HashMap::new();
+        let mut logs: Vec<Log> = Vec::new();
```

Comment:
```suggestion
        let mut logs = vec![];
```
See the other comment for the motivation of this suggestion. 👍